src/components/ChannelSearch/ChannelSearch.tsx

```diff
@@ -34,7 +34,13 @@ export type ChannelSearchProps<Us extends DefaultUserType<Us> = DefaultUserType>
   /** Custom UI component to display empty search results */
   SearchEmpty?: React.ComponentType;
   /** Custom search function to override default */
-  searchFunction?: (event: React.BaseSyntheticEvent) => Promise<void> | void;
+  searchFunction?: (
+    setQuery: React.Dispatch<React.SetStateAction<string>>,
+    setResults: React.Dispatch<React.SetStateAction<Array<UserResponse<Us>>>>,
+    setResultsOpen: React.Dispatch<React.SetStateAction<boolean>>,
+    setSearching: React.Dispatch<React.SetStateAction<boolean>>,
+    event: React.BaseSyntheticEvent,
```

Comment:
let's give this an object instead called `SearchFunctionParams` and define the type up top around line 20, then in the docs we can just say SearchFunctionParams instead of listing it all out